docs/basics/101-136-history.rst

```diff
@@ -0,0 +1,767 @@
+few chapters of the handbook. Find a commit :term:`shasum` in your history
+to specify the point in time you want to go back to:
+
+.. runrecord:: _examples/DL-101-136-121
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ git log -20 --oneline
+
+Let's go 15 commits back in time:
+
+.. runrecord:: _examples/DL-101-136-122
+   :language: console
+   :workdir: dl-101/DataLad-101
+   :realcommand: echo "$ git checkout $(git rev-parse HEAD~15)" && git checkout $(git rev-parse HEAD~15)
+
+How did your ``notes.txt`` file look at this point?
+
+.. runrecord:: _examples/DL-101-136-123
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ cat notes.txt
+
+Neat, isn't it? By checking out a commit shasum you can explore a previous
+state of a datasets history. And this does not only apply to simple text
+files, but every type of file in your dataset, regardless of size.
+The checkout command however led to something that Git calls a "detached HEAD state".
+While this sounds scary, a :command:`git checkout master` will bring you
+back into the most recent version of your dataset and get you out of the
+"detached HEAD state":
+
+.. runrecord:: _examples/DL-101-136-124
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ git checkout master
+
+
+Note one very important thing: The previously untracked files are still
+there.
+
+.. runrecord:: _examples/DL-101-136-125
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ datalad status
+
+The contents of ``notes.txt`` will now be the most recent version again:
+
+.. runrecord:: _examples/DL-101-136-126
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ cat notes.txt
+
+... Wow! You travelled back and forth in time!
+
+
+Undoing latest modifications of files
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Previously, we saw how to remove files from a datasets history that
+were accidentally saved and thus tracked for the first time.
+How does one undo a *modification* to a tracked file?
+
+Let's modify the saved ``Gitjoke1.txt``:
+
+.. runrecord:: _examples/DL-101-136-127
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ echo "this is by far my favourite joke!" >> Gitjoke2.txt
+
+.. runrecord:: _examples/DL-101-136-128
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ cat Gitjoke2.txt
+
+.. runrecord:: _examples/DL-101-136-129
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ datalad status
+
+.. runrecord:: _examples/DL-101-136-130
+   :language: console
+   :workdir: dl-101/DataLad-101
+
+   $ datalad save -m "add joke evaluation to joke" Gitjoke2.txt
+
+How could this modification to ``Gitjoke2.txt`` be undone?
+With the :command:`git reset` command again. If you want to
+"unsave" the modification but keep it in the file, use
+:command:`git reset --mixed` as before. However, if you want to
+get rid of the modifications entirely, use the option ``--hard``
+instead of ``--mixed``:
```

Comment:
Maybe point out that it only restores just the one file, because only that one file was modified. If the commit would have had other modifications, all of them would have been undone. This is one of the reasons why one should strive for commits to represent meaningful logical units of change.

Reply:
very good point, thx.